### PR TITLE
kernel-(build,install}.eclass: warn if USE=-initramfs

### PR DESCRIPTION
**Pull request by AndrewAmmerlaan**
as discussed on IRC @gentoo/dist-kernel

### CODE REVIEW

File: eclass/kernel-install.eclass [hunk 1]
```diff
@@ -549,6 +549,25 @@ kernel-install_pkg_pretend() {
 	fi
 }
 
+# @FUNCTION: kernel-install_pkg_pretend
+# @DESCRIPTION:
+# Print a warning if the initramfs use flag is not set
+kernel-install_pkg_setup() {
+	if ! use initramfs; then
```
Comment (thesamesam) [inline eclass/kernel-install.eclass:556]: I would perhaps only do it on new installs and/or first transition USE=initramfs->USE=-initramfs?

Comment (AndrewAmmerlaan) [reply eclass/kernel-install.eclass]: Good idea, fixed!

File: eclass/kernel-install.eclass [hunk 2]
```diff
@@ -549,6 +549,25 @@ kernel-install_pkg_pretend() {
 	fi
 }
 
+# @FUNCTION: kernel-install_pkg_pretend
```
Comment (mgorny) [inline eclass/kernel-install.eclass:552]: Wrong name.

File: eclass/kernel-install.eclass [hunk 3]
```diff
@@ -549,6 +549,25 @@ kernel-install_pkg_pretend() {
...
+# @DESCRIPTION:
+# Print a warning if the initramfs use flag is not set
+kernel-install_pkg_setup() {
```
Comment (mgorny) [inline eclass/kernel-install.eclass:555]: So why setup and not pretend?

**Review by mgorny — CHANGES_REQUESTED** [review]

File: eclass/kernel-install.eclass (hunk 3 above)
Comment (AndrewAmmerlaan) [reply eclass/kernel-install.eclass:555]: ooh my bad, it should have been pretend

File: eclass/kernel-install.eclass [hunk 4]
```diff
@@ -547,6 +547,20 @@ kernel-install_pkg_pretend() {
 			elog "    emerge --config ${CATEGORY}/${PN}:${SLOT}"
 		fi
 	fi
+
+	if ! use initramfs && ! has_version "=${CATEGORY}/${PN}[-initramfs]"; then
```
Comment (mgorny) [inline eclass/kernel-install.eclass:551]: That's not a valid atom.

Comment (AndrewAmmerlaan) [reply eclass/kernel-install.eclass]: Oops, that `=` is left over from when it was `PF` instead of `PN`.

**Review by mgorny — APPROVED** [review]
Good enough for me. I'm not exactly happy about spamming users with all these warnings but I suppose we don't really have a better solution.

**Pull request closed without merging**